core/taskengine/vm_runner_rest.go

```diff
@@ -115,9 +113,22 @@ func (r *RestProcessor) Execute(stepID string, node *avsproto.RestAPINode) (*avs
 		resp, err = request.Get(processedNode.Url)
 	}
 
-	response := string(resp.Body())
+	// Handle connection errors
+	if err != nil {
+		s.Success = false
+		s.Error = "HTTP request failed: connection error, timeout, or DNS resolution failure"
+		return s, fmt.Errorf("%s: %v", s.Error, err)
+	}
 
-	//maybeJSON := false
+	// Check HTTP status code - treat non-2xx/3xx as errors
+	if resp.StatusCode() < 200 || resp.StatusCode() >= 400 {
+		s.Success = false
+		errorMsg := fmt.Sprintf("Unexpected status code: %d", resp.StatusCode())
+		s.Error = errorMsg
+		return s, errors.New(errorMsg)
+	}
```

Comment:
This file failed the newly added test `TestRestRequestErrorHandling` because success is not update to false when there are invalid HTTP status code.

The change is to make sure success is set at the end, based on either `err` or `resp.statusCode()`